src/Common/TransformEndianness.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <base/Decimal_fwd.h>
+#include <base/extended_types.h>
+
+#include <utility>
+
+namespace DB
+{
+template <std::endian endian, typename T>
+    requires is_big_int_v<T>
+inline void transformEndianness(T & x)
+{
+    if constexpr (std::endian::native != endian)
+    {
+        std::ranges::transform(x.items, std::begin(x.items), [](auto& item) { return std::byteswap(item); });
+        std::ranges::reverse(x.items);
+    }
+}
+
+template <std::endian endian, typename T>
+    requires is_decimal<T> || std::is_floating_point_v<T>
+inline void transformEndianness(T & value)
+{
+    if constexpr (std::endian::native != endian)
+    {
+        auto * start = reinterpret_cast<std::byte *>(&value);
+        std::reverse(start, start + sizeof(T));
+    }
+}
+
+template <std::endian endian, typename T>
```

Comment:
(cosmetic: l. 32-38 are related to the big-integer case on l. 10-19. Could we sort it before l. 10-19?)